### ISSUE TITLE
Failure to start on EKS 1.27

### ISSUE THREAD

**Issue by danielloader** [XenitAB/spegel#190]
Hey just thought I'd file a bug report:

**Cluster**: v1.27.4-eks-2d98532
**Nodes**: v1.27.3-eks-a5565ad

**Log**

```
registry {"level":"error","ts":1693316449.014054,"caller":"build/main.go:71","msg":"","error":"Containerd registry config path is /etc/containerd/certs.d:/etc/docker/certs.d but needs to be /etc/containerd/certs.d for mirror configuration to take effect","stacktrace":"main.main\n\t/build/main.go:71\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
configuration {"level":"info","ts":1693316396.3836858,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://docker.io","path":"/etc/containerd/certs.d/docker.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3838172,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://ghcr.io","path":"/etc/containerd/certs.d/ghcr.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3839626,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://quay.io","path":"/etc/containerd/certs.d/quay.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3840818,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://mcr.microsoft.com","path":"/etc/containerd/certs.d/mcr.microsoft.com/hosts.toml"}
configuration {"level":"info","ts":1693316396.3842103,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://public.ecr.aws","path":"/etc/containerd/certs.d/public.ecr.aws/hosts.toml"}
configuration {"level":"info","ts":1693316396.3843098,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://gcr.io","path":"/etc/containerd/certs.d/gcr.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3844151,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://registry.k8s.io","path":"/etc/containerd/certs.d/registry.k8s.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3845265,"caller":"oci/containerd.go:286","msg":"added containerd mirror configuration","registry":"https://k8s.gcr.io","path":"/etc/containerd/certs.d/k8s.gcr.io/hosts.toml"}
configuration {"level":"info","ts":1693316396.3846445,"caller":"build/main.go:74","msg":"gracefully shutdown"}
```

**Flux Config to Deploy**

```
---
apiVersion: v1
kind: Namespace
metadata:
  name: spegel
  labels:
    toolkit.fluxcd.io/tenant: sre-team
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: HelmRepository
metadata:
  name: spegel
  namespace: spegel
spec:
  type: "oci"
  interval: 5m0s
  url: oci://ghcr.io/xenitab/helm-charts
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: spegel
  namespace: spegel
spec:
  interval: 1m
  chart:
    spec:
      chart: spegel
      version: "v0.0.11"
      interval: 5m
      sourceRef:
        kind: HelmRepository
        name: spegel
```

Deploy is failing with the same error log as above on both managed node group nodes and karpenter provisioned spot instances.

**Comment by noyoshi** [comment]
I am also having this issue, but we are on a different version: `v1.24.15-eks-a5565ad`.

**Comment by phillebaba** [comment]
The road to hell is paved with good intentions...

I introduced a new verify features in #158 because I was observing users having issues with incorrect config paths while wondering why Spegel is not working. Turns out the `ConfigPath` is not always a single path but can also be multiple paths separated by a `:`. Here is an example of how Containerd deals with it.
https://github.com/containerd/containerd/blob/b0a7f08aa7490a3d43787b9f2b444e4b20b96f8c/pkg/cri/server/image_pull.go#L364C30-L364C58

I will cut a new release as soon as I have a fix for this.
